libvirt/tests/src/virtual_network/iface_hotplug.py

```diff
@@ -192,17 +192,22 @@ def get_all_mac_in_vm():
                         test.fail("Failed to attach-interface: %s" % ret.stderr.strip())
                 elif stress_test:
                     if attach_device:
-                        # Detach the device immediately for stress test
-                        ret = virsh.detach_device(vm_name, iface_xml_obj.xml,
-                                                  flagstr=detach_option,
-                                                  ignore_status=True)
+                        # Detach the device for stress test
+                        utils_misc.wait_for(lambda: virsh.detach_device(vm_name,
+                                                                        iface_xml_obj.xml,
+                                                                        flagstr=detach_option,
+                                                                        ignore_status=False,
+                                                                        debug=True),
+                                            timeout=9, first=3, step=1)
                     elif attach_iface:
                         # Detach the device immediately for stress test
                         options = ("--type %s --mac %s %s" %
                                    (iface_type, mac, detach_option))
-                        ret = virsh.detach_interface(vm_name, options,
-                                                     ignore_status=True)
-                    libvirt.check_exit_status(ret)
+                        utils_misc.wait_for(lambda: virsh.detach_interface(vm_name,
+                                                                           options,
+                                                                           ignore_status=False,
+                                                                           debug=True),
+                                            timeout=9, first=3, step=1)
```

Comment:
libvirt.check_exit_status(ret) should not be deleted. Because we need to check the results of virsh detach-interface/device and raise proper exceptions if need.


```suggestion
                       ret = utils_misc.wait_for(lambda: virsh.detach_interface(vm_name,
                                                                           options,
                                                                           debug=True),
                                            timeout=9, first=3)
                   libvirt.check_exit_status(ret)
```